internal/runtime/client/client.go

```diff
@@ -655,3 +659,20 @@ func (c *client) matchNamespace(ctx context.Context, selector labels.Selector, n
 
 	return selector.Matches(labels.Set(ns.GetLabels())), nil
 }
+
+// NameForHandler constructs a canonical name for a registered runtime extension handler.
+func NameForHandler(handler runtimehooksv1.ExtensionHandler, extensionConfig *runtimev1.ExtensionConfig) (string, error) {
```

Comment:
Added a couple of utility methods and unit tests in the client package to express and enforce the naming convention for discovered handlers.

This was already enforced by validation of the extension name and the handler name in webhooks. Neither name can contain a dot by design, so they should always be able to be combined and split this way.

Reply:
Thx!